homeassistant/components/konnected/__init__.py

```diff
@@ -430,20 +465,21 @@ def binary_value(state, activation):
         if device is None:
             return self.json_message('unregistered device',
                                      status_code=HTTP_BAD_REQUEST)
-        pin_data = device[CONF_BINARY_SENSORS].get(pin_num)
+        pin_data = device[CONF_BINARY_SENSORS].get(pin_num) or \
+            next((s for s in device[CONF_SENSORS] if s[CONF_PIN] == pin_num),
+                 None)
 
         if pin_data is None:
             return self.json_message('unregistered sensor/actuator',
                                      status_code=HTTP_BAD_REQUEST)
 
-        entity_id = pin_data.get(ATTR_ENTITY_ID)
-        if entity_id is None:
-            return self.json_message('uninitialized sensor/actuator',
-                                     status_code=HTTP_NOT_FOUND)
-        state = bool(int(state))
-        if pin_data.get(CONF_INVERSE):
-            state = not state
+        pin_data['device_id'] = device_id
+
+        for attr in ['state', 'temp', 'humi', 'addr']:
+            value = payload.get(attr)
+            if value is not None:
+                handler = HANDLERS.get(attr)
+                if handler:
+                    hass.async_create_task(handler(hass, pin_data, payload))
```

Comment:
Is it better to just create new tasks than awaiting the tasks here?

Reply:
I don't know. It probably doesn't really matter. To be honest I was just mimicking patterns in other parts of the code.

Reply:
I think it's fine like this.